### PR TITLE
Collapse expanded panels when setting singleBranchExpand to true 12.3.x

### LINKED ISSUE #10509: IgxAccordion singleBranchExpand should collapse the expanded panels
Having several panels expanded and setting singleBranchExpand to true should collapse all panels except the last expanded one. Currently it persists all the expanded panels.

### PR DESCRIPTION
**Pull request by mmart1n**
Closes #10509   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by wnvko — CHANGES_REQUESTED** [review]
Setting `singleBranchExpand` to true is collapsing all panels except the disabled ones. What should happen if at this point you enable given panel - should it collapse or the other expanded panel should collapse?

**Comment by mmart1n** [comment]
> Setting `singleBranchExpand` to true is collapsing all panels except the disabled ones. What should happen if at this point you enable given panel - should it collapse or the other expanded panel should collapse?

We haven't discussed this case yet. @StefanIvanov Any recommendations?

**Comment by StefanIvanov** [comment]
@mmart1n yes I think we should leave the handling to the developer since the described case by @wnvko is possible only via the API, there is no scenario that I can come up with happening via direct user interaction.

**Review by wnvko — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/accordion/accordion.component.ts [hunk 1]
```diff
@@ -256,6 +269,16 @@ export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, O
         this.panels.forEach(panel => panel.collapse());
     }
 
+    private collapseAllExceptLast(): void {
+        const lastExpanded = this.panels?.filter(p => !p.collapsed && !p.header.disabled).pop();
+        this.panels?.forEach((p: IgxExpansionPanelComponent) => {
+            if (p !== lastExpanded && !p.header.disabled) {
+                p.collapsed = true;
+                p.cdr.detectChanges();
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/accordion/accordion.component.ts:277]: One call to `cdr.detectChanges()` after collapsing all panels would be better than calling `detectChanges()` on each collapsed panel. Don't reference each expansion panel `cdr` but instead call detectChanges on a cdr from the accordion itself after the `forEach` loop.

**Review by kdinev — APPROVED** [review]

File: projects/igniteui-angular/src/lib/accordion/accordion.component.ts (hunk 1 above)
Comment (kdinev) [inline projects/igniteui-angular/src/lib/accordion/accordion.component.ts:277]: One call to `cdr.detectChanges()` after collapsing all panels would be better than calling `detectChanges()` on each collapsed panel. Don't reference each expansion panel `cdr` but instead call detectChanges on a cdr from the accordion itself after the `forEach` loop.

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts [hunk 2]
```diff
@@ -199,7 +199,7 @@ export class IgxExpansionPanelComponent extends ToggleAnimationPlayer implements
 
     private _collapsed = true;
 
-    constructor(private cdr: ChangeDetectorRef, protected builder: AnimationBuilder, private elementRef?: ElementRef) {
+    constructor(public cdr: ChangeDetectorRef, protected builder: AnimationBuilder, private elementRef?: ElementRef) {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts:202]: You can move this back to `private`

File: projects/igniteui-angular/src/lib/accordion/accordion.component.ts [hunk 3]
```diff
@@ -256,6 +269,16 @@ export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, O
         this.panels.forEach(panel => panel.collapse());
     }
 
+    private collapseAllExceptLast(): void {
+        const lastExpanded = this.panels?.filter(p => !p.collapsed && !p.header.disabled).pop();
+        this.panels?.forEach((p: IgxExpansionPanelComponent) => {
+            if (p !== lastExpanded && !p.header.disabled) {
+                p.collapsed = true;
+            }
+            this.cdr.detectChanges();
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/accordion/accordion.component.ts:278]: Here you still run `detectChanges()` on every iteration. Move it outside of the `forEach` loop so it's ran once after all iterations.

**Review by kdinev — CHANGES_REQUESTED** [review]

**Review by kdinev — APPROVED** [review]

**Pull request merged**